#167897145 Setup babel

mthamayor opened the pull request:
#### What does this PR do?
Setup babel transpiler

#### Description of Task to be completed?
- Set up babel transpiler to enable the use of es6+ features
- Add start, development and build scripts
- Update README.md to reflect new changes

#### How should this be manually tested?
- CD into the root folder
- Run `npm run build` in your CLI. This will generate a **dist** folder that contains the transpiled project 
- You can run `npm run start` to start up the transpiled version of the application
- You can run `npm run dev` to start up the development server

#### Any background context you want to provide?
You need to have npm version `5.2+` or greater installed as the scripts are run with `npx` which requires the aforementioned npm version or higher

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/167897145

#### Screenshots (if appropriate)
None

#### Questions:
None

mthamayor committed:
chore(setup-babel):install and configure babel
[Finishes #167897145]

okabamac commented on babel.config.js:
```diff
@@ -0,0 +1,11 @@
+  ],
+];
+
+module.exports = { presets };
```
This looks great to me. Great job

mthamayor replied:
Hello, Thank you for your feedback. Babel recommends babel.config.js for configuration. You can check this configuration page https://babeljs.io/docs/en/configuration

sebaztyn reviewed (approved):
This is fine.

Stylll reviewed (commented):
Hey @mthamayor Good Job!. Please resolve the merge conflicts.

Stylll commented on package.json:
```diff
@@ -11,6 +11,7 @@
   "license": "MIT",
```
hey @mthamayor, I noticed you removed these script commands you added before. Is there a reason why?

mthamayor replied:
Hello, thank you for bring my attention to this. It's a mistake on my end. I'll work on this as soon as possible.

mthamayor committed:
chore(babel-setup):add package lock to gitignore

mthamayor committed:
chore(delete-package-lock):delete package-lock

mthamayor committed:
chore(setup-babel):resolve merge conflict

mthamayor committed:
chore(setup-babel):change .package-lock.json to package-lock.json

mthamayor committed:
chore(setup-babel):gitignore fix

mthamayor committed:
chore(setup-babel):add babel support

- add development script
- Update README.md to reflect new script actions
- Add dist directory to gitignore

Stylll commented on babel.config.js:
```diff
@@ -0,0 +1,13 @@
+/* eslint linebreak-style: ["error", "windows"] */
```
Hey @mthamayor , I don't think this line is necessary here. You may want to remove it. I believe that is why the hound bot is giving you an error about linebreaks.

mthamayor replied:
Hello, I have worked on it as requested

Stylll reviewed (changes requested):
Great job @mthamayor . There's one more change required. Kindly look into it.

mthamayor committed:
chore(setup-babel):remove inline lint override

mthamayor committed:
chore(setup-babel):update readme vision message

mthamayor committed:
chore(setup-babel):update readme header message
